collections/map.go

```diff
@@ -7,16 +7,28 @@ import (
 	storetypes "github.com/cosmos/cosmos-sdk/store/types"
 )
 
-// NewMap returns a Map given a StoreKey, a Prefix and the relative value and key encoders.
+// NewMap returns a Map given a StoreKey, a Prefix, human-readable name and the relative value and key encoders.
+// Name and prefix must be unique within the schema and name must match the format specified by NameRegex, or
+// else this method will panic.
 func NewMap[K, V any](
-	sk storetypes.StoreKey, prefix Prefix,
+	schema Schema, prefix Prefix, name string,
+	keyCodec KeyCodec[K], valueCodec ValueCodec[V],
```

Comment:
nit: generally I'd prefer to be consistent and have argument-per-line grouped by type

```suggestion
	schema Schema,
	prefix Prefix,
	name string,
	keyCodec KeyCodec[K],
	valueCodec ValueCodec[V],
```